### PR TITLE
vo_opengl: fancy-downscaling: enable also for anamorphic clips

### PR DESCRIPTION
**Pull request by avih**
Sub optimal, but since it only improves on the original kernel and never regresses the scaling, it's better than not respecting `fancy-downscaling` at all for anamorphic clips.

### CODE REVIEW

**Comment by haasn** [comment]
Longer-term it would be possible to make fancy-downscaling work independently in both directions (for convolution filters, you'd have two LUTs - one per axis, and for elliptic filters you'd filter from an ellipse instead of a circle), but this would require more invasive code changes for comparatively little benefit.

**Comment by avih** [comment]
(The updated PR adds DOCs)

**Comment by avih** [comment]
It's _probably_  possible (depending on what's special with `FFMAX`) to simplify
`scale_factor = FFMAX(1.0, 1.0 / f);`
to
`scale_factor = 1.0 / f;`
Since we know `f < 1.0` already, therefore `1.0 / f` should be guaranteed to be bigger than `1.0` for `FFMAX`.

**Comment by avih** [comment]
@lachs0r , do you recall the case where it sucked? if yes, could you check it again with this patch and see if it's still bad?

**Comment by avih** [comment]
FWIW, I've been using this patch frequently with downscaled anamorphic clips, and IMO it feels perfect and definitely better than without it when I request and expect fancy downscale.

**Comment by haasn** [comment]
Then I'd tentatively +1 unless @lachs0r can recall/reproduce a clip where it sucked.

@avih One thing you could test is using a reference “downscaling” clip and manually turning it into an anamorphic image (eg. using imagemagick's -resize operator), then seeing how well that downscales compared to the original.

(Disregarding any inherent differences between the anamorphic ver and the original)

**Comment by avih** [comment]
Optimal fancy (using @haasn `rings_lg_orig.png` displayed as 16:9 in mpv, then took a screenshot, and fancy-downscale this screenshot):


Using the patch in this PR, (using the original `rings_lg_orig.png` displayed as 16:9 and downscaled to 15%):


And finally using the same as above but without the patch:

**Comment by haasn** [comment]
This is very obviously a strict improvement, it adds no artifacts that would not have been considerably worse without the patch - only diminishes them.

**Comment by wm4** [comment]
So, uh, do we want this in this form?

Does anyone want to implement it "properly", since generating and using different LUTs shouldn't be that hard anymore?

>This is very obviously a strict improvement,

If you say this, it actually sounds pretty good. But isn't it still incorrect?

**Comment by haasn** [comment]
> Does anyone want to implement it "properly", since generating and using different LUTs shouldn't be that hard anymore?

I don't think this is worth the effort, since *who* watches anamorphic downscaled clips and cares about fancy-downscaling? @avih might be the only person desiring such a feature. Generating and using different LUTs shouldn't be too hard, but it would still be a lot of lines of code (relatively speaking, compared to this patch).

> But isn't it still incorrect?

Yes. Though it's no *less* incorrect than the old logic, which was simply causing significant aliasing in all directions.

The bottom line is, I think, the fact that this an almost literal one-line change (excluding documentation and comments) with no side effects, so I consider it a harmless improvement in lieu of somebody wanting to write a full fix.

**Comment by wm4** [comment]
Somehow overlooked haasn's reply. Merged, thanks.
